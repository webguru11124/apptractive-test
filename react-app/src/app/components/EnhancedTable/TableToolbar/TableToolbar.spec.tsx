import React from 'react';
import { test } from 'vitest';
import { render, cleanup } from '@testing-library/react';
import { EnhancedTableToolbar } from './TableToolbar';

afterEach(cleanup);

describe('EnhancedTableToolbar Component', () => {
  test('renders "Invoices" title when no selection', () => {
    const { queryByText } = render(<EnhancedTableToolbar numSelected={0} />);
    const title = queryByText('Invoices');
    expect(title).not.toBeNull();
  });

  test('renders "selected" title when items selected', () => {
    const numSelected = 3;
    const { queryByText } = render(
      <EnhancedTableToolbar numSelected={numSelected} />
    );
    const title = queryByText(`${numSelected} selected`);
    expect(title).not.toBeNull();
  });
});
