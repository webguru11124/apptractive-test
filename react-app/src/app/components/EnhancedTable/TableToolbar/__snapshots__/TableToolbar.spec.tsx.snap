// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EnhancedTableToolbar Component > rendered successfully 1`] = `
<body>
  <div>
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-pjjt8i-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 css-2pbtu6-MuiTypography-root"
        data-testid="invoices"
        id="tableTitle"
      >
        invoices
      </div>
      <button
        aria-label="filterList"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element="true"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FilterListIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
